pkg/applier/applier.go

```diff
@@ -150,77 +152,56 @@ var _ Supervisor = &supervisor{}
+	syncKind := scope.SyncKind()
+	syncNamespace := scope.SyncNamespace()
 	invInfo := inventory.NewSingleObjectInfo(
 		inventory.ID(InventoryID(syncName, syncNamespace)),
 		types.NamespacedName{
 			Name:      syncName,
 			Namespace: syncNamespace,
 		},
 	)
+	policy := inventory.PolicyAdoptIfNoInventory
+	if scope == declared.RootScope {
+		policy = inventory.PolicyAdoptAll
+	}
 	a := &supervisor{
 		invInfo:          invInfo,
 		clientSet:        cs,
-		policy:           inventory.PolicyAdoptIfNoInventory,
+		policy:           policy,
 		syncKind:         syncKind,
 		syncName:         syncName,
 		syncNamespace:    syncNamespace,
 		reconcileTimeout: reconcileTimeout,
 	}
-	klog.V(4).Infof("Namespace Supervisor %s/%s is initialized", namespace, syncName)
-	return a, nil
+	klog.V(4).Infof("%s Supervisor %s/%s is initialized", syncKind, syncNamespace, syncName)
+	return a
 }
 
-// NewRootSupervisor constructs a Supervisor that can manage both cluster-level
-// and namespace-level resource objects in a single cluster.
-func NewRootSupervisor(cs *ClientSet, syncName string, reconcileTimeout time.Duration) (Supervisor, error) {
-	syncKind := configsync.RootSyncKind
-	syncNamespace := configsync.ControllerNamespace
-	u := newInventoryUnstructured(syncName, syncNamespace)
-	// If the ResourceGroup object exists, annotate the status mode on the
-	// existing object.
-	if err := annotateStatusMode(context.TODO(), cs.Client, u, cs.StatusMode); err != nil {
-		klog.Errorf("failed to annotate the ResourceGroup object with the status mode %s", cs.StatusMode)
-		return nil, err
-	}
-	klog.Infof("successfully annotate the ResourceGroup object with the status mode %s", cs.StatusMode)
-	invInfo := inventory.NewSingleObjectInfo(
-		inventory.ID(InventoryID(syncName, syncNamespace)),
-		types.NamespacedName{
-			Name:      syncName,
-			Namespace: syncNamespace,
-		},
-	)
-	a := &supervisor{
-		invInfo:          invInfo,
-		clientSet:        cs,
-		policy:           inventory.PolicyAdoptAll,
-		syncKind:         syncKind,
-		syncName:         syncName,
-		syncNamespace:    syncNamespace,
-		reconcileTimeout: reconcileTimeout,
-	}
-	klog.V(4).Infof("Root Supervisor %s is initialized and synced with the API server", syncName)
-	return a, nil
+// AnnotateStatusMode sets the status mode annotation on the ResourceGroup object
+// if it exists.
+// This is invoked once before the applier runs so that ResourceGroups which are
+// too large can have their status cleared by the resource-group-controller.
+func (s *supervisor) AnnotateStatusMode(ctx context.Context) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		u := newInventoryUnstructured(s.syncName, s.syncNamespace)
+		err := s.clientSet.Client.Get(ctx, client.ObjectKey{Name: u.GetName(), Namespace: u.GetNamespace()}, u)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
```

Comment:
nit: add comment: If the ResourceGroup doesn't exist yet, cli-utils will create it with the right status policy when it starts the next apply/destroy.